### PR TITLE
Bounce ends when marker is clicked

### PR DESCRIPTION
**Pull request by MikeDuister**
## Description
Bounce animation stops when marker is clicked. 

## Motivation and Context
Prevents continuous loop of bouncing pokémon ( #1955 )

**rework of #1979 based on improvements in #2017**

## How Has This Been Tested?
By running the setup several times on Linux and Windows.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by MikeDuister** [comment]
# request
Late night variable naming might not have been the best idea, please report if variables need to be renamed.

**Comment by clbuttic** [comment]
Why not time them out? 

```
            setTimeout(function() {
                marker.setAnimation(null);
            }, 4000);
```

**Comment by louispires** [comment]
@clbuttic Please see comments here: #1979

Basically timing it out will be useless for some people as there is no certain way of knowing that the user has noticed the bouncing animation.

File: static/map.js [hunk 1]
```diff
@@ -497,13 +497,18 @@ function getGoogleSprite(index, sprite, display_height) {
...
-function setupPokemonMarker(item, skipNotification) {
+function setupPokemonMarker(item, skipNotification, isBounceDisabled) {
 
     // Scale icon size up with the map exponentially
     var icon_size = 2 + (map.getZoom()-3) * (map.getZoom()-3) * .2 + Store.get('iconSizeModifier');
     var pokemon_index = item.pokemon_id - 1;
     var sprite = pokemon_sprites[Store.get('pokemonIcons')] || pokemon_sprites['highres']
     var icon = getGoogleSprite(pokemon_index, sprite, icon_size);
+	
+	var animationDisabled = false;
```
Comment (invisiblek) [inline static/map.js]: tabs vs spaces

**Comment by clbuttic** [comment]
@louispires I see. Is it not going to be too annoying? Wouldn't it be better to show some other indicator that also incorporates time left? Either fading the opacity, or a spinner.

**Comment by MikeDuister** [comment]
@clbuttic Notifications about pokémon are meant for rare pokémon that the person is willing to travel for. With the default radius this means that about 5 pokémon will be bouncing at max.

But even if the animation needs to be changed, it will be in another separate PR. This issue stand aside from this fix.

File: static/map.js [hunk 2]
```diff
@@ -514,7 +519,13 @@ function setupPokemonMarker(item, skipNotification) {
         optimized: false,
         map: map,
         icon: icon,
+		animationDisabled: animationDisabled,
```
Comment (invisiblek) [inline static/map.js]: animationDisabled: isBounceDisabled

File: static/map.js (hunk 1 above)
Comment (invisiblek) [inline static/map.js]: this variable is useless, just use the parameter passed in (see comment below)

File: static/map.js [hunk 3]
```diff
@@ -860,8 +872,12 @@ function updateMap() {
 function redrawPokemon(pokemon_list) {
     var skipNotification = true;
     $.each(pokemon_list, function(key, value) {
+		var isBounceDisabled = false;
+		if(this.marker.animationDisabled != false){
+			isBounceDisabled = true;
+		};
         var item =  pokemon_list[key];
-        var new_marker = setupPokemonMarker(item, skipNotification);
+        var new_marker = setupPokemonMarker(item, skipNotification, isBounceDisabled);
```
Comment (invisiblek) [inline static/map.js]: can you just send this.marker.animationDisabled? no need to define more variables for all this stuff imo

**Comment by MikeDuister** [comment]
@invisiblek Ff the default isn't defined by this variable it will return undefined at first run, isn't this an issue?

**Comment by AHAAAAAAA** [comment]
Tested, works +1

**Comment by Idhanta** [comment]
The bounce doesn't stop after I click on the marker. Using safari.

**Pull request merged**